### PR TITLE
Feature/busd

### PR DESCRIPTION
**Pull request by micahriggan**
This needs to go, bump version, update copay mr

### CODE REVIEW

**Comment by justinkook** [comment]
`bitcore-wallet-service/src/lib/pushnotificationsservice.ts`

Add BUSD

```
_getDataForTemplate(notification: INotification, recipient, cb) {
    const UNIT_LABELS = {
      btc: 'BTC',
      bit: 'bits',
      bch: 'BCH',
      eth: 'ETH',
      xrp: 'XRP',
      usdc: 'USDC',
      pax: 'PAX',
      gusd: 'GUSD',
      busd: 'BUSD'
    };
```

File: packages/bitcore-wallet-client/src/lib/common/constants.ts [hunk 1]
```diff
@@ -110,10 +121,16 @@ export const Constants = {
       }
     }
   },
-  COINS: ['btc', 'bch', 'eth', 'xrp', 'usdc', 'pax', 'gusd'],
-  ERC20: ['usdc', 'pax', 'gusd'],
+  COINS: ['btc', 'bch', 'eth', 'xrp', 'usdc', 'pax', 'gusd', 'busd'],
+  ERC20: ['usdc', 'pax', 'gusd', 'busd'],
   UTXO_COINS: ['btc', 'bch'],
   TOKEN_OPTS: {
+    '0x4Fabb145d64652a948d72533023f6E7A623C7C53': {
```
Comment (micahriggan) [inline packages/bitcore-wallet-client/src/lib/common/constants.ts:128]: needs to be lowercase

File: packages/bitcore-wallet-service/src/lib/pushnotificationsservice.ts [hunk 2]
```diff
@@ -338,16 +338,24 @@ export class PushNotificationsService {
...
     if (data.amount) {
       try {
-        const unit = data.tokenAddress
-          ? Constants.TOKEN_OPTS[data.tokenAddress.toLowerCase()].symbol.toLowerCase()
-          : recipient.unit.toLowerCase();
-        data.amount = Utils.formatAmount(+data.amount, unit) + ' ' + UNIT_LABELS[unit];
+        let unit = recipient.unit.toLowerCase();
+        let label = UNIT_LABELS[unit];
+        if (data.tokenAddress) {
+          if (Constants.TOKEN_OPTS[data.tokenAddress.toLowerCase()]) {
```
Comment (micahriggan) [inline packages/bitcore-wallet-service/src/lib/pushnotificationsservice.ts:351]: `let tokenAddress = data.tokenAddress.toLowerCase()`

File: packages/bitcore-wallet-client/src/lib/common/constants.ts [hunk 3]
```diff
@@ -110,10 +121,16 @@ export const Constants = {
...
+      name: 'Binance USD Coin',
+      symbol: 'BUSD',
+      decimal: 18,
+      address: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'
```
Comment (micahriggan) [inline packages/bitcore-wallet-client/src/lib/common/constants.ts]: wrong address

File: packages/bitcore-wallet-service/package.json [hunk 4]
```diff
@@ -28,6 +28,7 @@
     "bitcore-lib-cash": "^8.17.0",
     "body-parser": "^1.11.0",
     "compression": "^1.6.2",
+    "bitcore-wallet-client": "^8.17.0",
```
Comment (matiu) [inline packages/bitcore-wallet-service/package.json:31]: no, please remove BWC deps from BWS.

BWC already depends on BWS for testing.

Comment (matiu) [reply packages/bitcore-wallet-service/package.json:31]: we could move them to CWC.

Comment (micahriggan) [reply packages/bitcore-wallet-service/package.json:31]: That seems backwards to me, but I can remove it.

I would normally have the client required in the server.

File: packages/bitcore-wallet-client/src/lib/api.ts [hunk 5]
```diff
@@ -54,6 +54,7 @@ export class API extends EventEmitter {
   bp_partner: string;
   bp_partner_version: string;
 
+  static Constants = Constants;
```
Comment (matiu) [inline packages/bitcore-wallet-client/src/lib/api.ts:57]: remove this?

File: packages/bitcore-wallet-service/src/lib/pushnotificationsservice.ts [hunk 6]
```diff
@@ -338,16 +338,25 @@ export class PushNotificationsService {
...
-        data.amount = Utils.formatAmount(+data.amount, unit) + ' ' + UNIT_LABELS[unit];
+        let unit = recipient.unit.toLowerCase();
+        let label = UNIT_LABELS[unit];
+        if (data.tokenAddress) {
+          const tokenAddress = data.tokenAddress.toLowerCase();
+          if (Constants.TOKEN_OPTS[tokenAddress]) {
+            unit = Constants.TOKEN_OPTS[tokenAddress].symbol.toLowerCase();
+            label = UNIT_LABELS[unit];
+          } else {
+            label = 'tokens';
```
Comment (matiu) [inline packages/bitcore-wallet-service/src/lib/pushnotificationsservice.ts:356]: nice.

**Pull request merged**